### ISSUE TITLE
Reference force field to rerun force field

### ISSUE THREAD

**Issue by ramess101** [ramess101/MBAR_GCMC#12]
@mrshirts @msoroush @jpotoff

I have repeated the simulations that Mick et al. ran for 2-methylpropane, 2,2-dimethylpropane and 2,2,4-trimethylpentane. The goal of this exercise was to run the GCMC simulations using TraPPE or Potoff and then try to predict the other force field saturation curve from this configurations, i.e., predict Potoff from TraPPE simulations and predict TraPPE from Potoff simulations. The lines in the figure below are from the direct simulations while the points are from the MBAR predictions. Red are for Potoff and blue are for TraPPE.

![image](https://user-images.githubusercontent.com/23408516/48293038-4c0da980-e43a-11e8-9261-5351aae6d3d8.png)

The main takeaway from this figure is that predicting Potoff from TraPPE or TraPPE from Potoff works relatively well for the vapor densities, vapor pressure, and enthalpy of vaporization. However, the saturated liquid densities are very sporadic. This is not ideal from a force field optimization stand point, but it could help explore different values of lambda when trying to improve the TraPPE parameters such that they match vapor pressure. 

Recall that using MBAR with ITIC performed much worse at predicting all of these properties when changing lambda from 12 to 16. For a constant lambda, however, MBAR should be very accurate for the GCMC approach, similar to what we saw for ITIC.

**Comment by mrshirts** [comment]
> However, the saturated liquid densities are very sporadic.

What are the N_eff there? It would be good to understand why they are off.

**Comment by ramess101** [comment]
@mrshirts 

As expected, this is an overlap issue, manifested by the low number of effective samples (N_eff). For these large changes in force fields, the Neff of the liquid phase are typically less than 10. By contrast, the Neff of the vapor phase is around 100,000. Obviously the exact values depend on how many frames are stored. But the extremely low efficiency of the liquid phase samples suggests that brute force (i.e., just sampling longer) is clearly not the right way to go. I think varying mu, T, or sigma would be a much more efficient alternative.

**Comment by mrshirts** [comment]
Yep, liquid should have a much harder time of overlap, since there are so many more potential intrusions into high E energy areas of the potential surface.  Varying those parameters is the way to go, I think.

**Comment by ramess101** [comment]
@mrshirts @msoroush @jpotoff

Here are the results where we predict force field j with configurations sampled with force field i. The top figure is where the repulsive exponent is constant (either LJ 12-6 or Mie 16-6) while the bottom figure is where the exponent varies (from 12-6 to 16-6 and vice versa):

![image](https://user-images.githubusercontent.com/23408516/49096847-ad9a8a00-f228-11e8-9447-73c01ebd4eaa.png)

![image](https://user-images.githubusercontent.com/23408516/49096855-b4290180-f228-11e8-970e-5e7bba4df71f.png)

The main conclusions are:

- Vapor properties are predicted much more accurately than liquid properties
- Converting between lambdas is worse for liquid properties but still good for vapor properties

The question remains as to why TraPPE -> NERD is much less reliable for 2,3,4-trimethylpentane. Currently we think this is because of the difference in the X-CH-X angle parameters for TraPPE and NERD. But it is still surprising that this would impact the vapor phase estimates as well.

**Comment by mrshirts** [comment]
I think generally the thing to always look at is what the N_eff is for the bad predictions; that seems so far to be the determining factor.  Then we can figure out why.  

Reweighting does seem to get the answer mostly right: any iterative process for optimization (i.e. carry out a new simulation at the predicted point, redo reweighting) seems likely to work well.

**Comment by ramess101** [comment]
@mrshirts @msoroush @jpotoff

I agree that the number of effective samples is the key to determine when a prediction is good or not. This plot shows the number of effective samples for the vapor and liquid phases when converting between TraPPE and MiPPE-gen (i.e., the more challenging problem where lambda changes between 12 and 16)

![image](https://user-images.githubusercontent.com/23408516/49103845-ae87e780-f239-11e8-80f2-d70d32a3c71e.png)

The heuristic of 50 samples suggests that none of the liquid phases have enough samples to have much confidence in the predicted liquid properties, while the vapor properties should be quite accurate.

**Comment by ramess101** [comment]
@mrshirts @msoroush @jpotoff

What do people think about this figure? I think it shows how the liquid and vapor number of effective samples (now labeled Keff_snaps) are quite different. It also shows why MiPPE-gen to MiPPE-SL is accurate in liquid phase as Keff_snaps is much greater than 50, while this is only the case for one of the TraPPE to NERD systems (2-methylpropane).

![image](https://user-images.githubusercontent.com/23408516/49112974-7db3ac80-f251-11e8-863e-ac1ea25b3cf7.png)

**Comment by ramess101** [comment]
@mrshirts @msoroush @jpotoff

Another option is a percent deviation plot like this one:

![image](https://user-images.githubusercontent.com/23408516/49113651-56f67580-f253-11e8-9e36-32101389abc9.png)

The problem with this plot is that we only have a few points at very high Keff_snaps and several points at very low Keff_snaps, but not much in between. I think the easiest way to generate more points is to just reprocess the systems with high Keff_snaps but taking a systematic subset of the data, i.e., first 90% of the data, then 80% of the data, etc. so that we get some Keff_snaps in the intermediate range.

Do people think we should include a plot like this one after filling in the rest of the Keff_snaps space?
